controller.rb

```diff
@@ -0,0 +1,38 @@
+require_relative 'view'
+require_relative 'flashcard'
+require_relative 'deck'
+require_relative 'card_parser'
+
+require 'pry'
+
+class Controller
+  include Parsable
+  attr_reader :new_deck
+
+  def initialize
+    @new_deck = Deck.new(parse_from_txt('source_cards.txt'))
+    @view = View.new
+    @view.introduction
+    @view.user_interface
+    run_interface
+  end
+
+  def run_interface
+    random_number = rand(0..@new_deck.flashcards.length)
+    entry = @view.get_input
+    if entry == 'yes'
+      @view.question(@new_deck.flashcards, random_number)
+      until @view.get_input == "#{@new_deck.flashcards[random_number].term}"
```

Comment:
Again, here, the controller knows too much about about the innards of the deck. I think the Deck should have a method that accepts an argument that is the user's guess and then the deck will tell the controller whether that was right or wrong.

```ruby
deck.correct_answer_for_current_card?(user_input)
```